arrows/pytorch/data_augmentation.py

```diff
@@ -0,0 +1,309 @@
+# ckwg +28
+# Copyright 2018 by Kitware, Inc.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#  * Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+#  * Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+#  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+#    to endorse or promote products derived from this software without specific
+#    prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+import sys
+import torch
+import pickle
+
+from torchvision import models, transforms
+from torch.autograd import Variable
+from torch import nn
+
+import numpy as np
+import scipy as sp
+import scipy.optimize
+import threading
+
+from PIL import Image as pilImage
+
+from sprokit.pipeline import process
+from sprokit.pipeline import datum
+from kwiver.kwiver_process import KwiverProcess
+from vital.types import Image
+from vital.types import DetectedObject, DetectedObjectSet
+from vital.types import new_descriptor, DescriptorSet
+
+from timeit import default_timer as timer
+
+from vital.util.VitalPIL import get_pil_image
+
+from kwiver.arrows.pytorch.grid import Grid
+from kwiver.arrows.pytorch.augmented_resnet_feature_extractor import AugmentedResnetFeatureExtractor
+
+def to_vital(raw_data):
+    if len(raw_data) == 0:
+        return DescriptorSet()
+    vital_descriptors = []
+    for item in raw_data:
+        new_desc=new_descriptor(item.size)
+        new_desc[:]=item
+        vital_descriptors.append(new_desc)
+    return DescriptorSet(vital_descriptors)
+
+class DataAugmentation(KwiverProcess):
```

Comment:
There should probably be more documentation here, especially the configuration parameters.